docs/src/components/ColorSwatch/ColorSwatch.component.tsx

```diff
@@ -0,0 +1,104 @@
+// REACT
+import * as React from 'react';
+// VENDOR
+import styled, { css } from 'styled-components';
+import classNames from 'classnames';
+import { Pill } from '../Pill'
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+// THEME
+import { colors, sizes } from '../../../../src/theme';
+
+
+// INTERFACES
+interface ColorSwatchProps {
+  className?: string;
+  backgroundColor: string;
+  contrastOnBlack: number;
+  contrastOnF1F1F1: string;
+  contrastOnFAFAFA: string;
+  contrastOnWhite: number;
+  copyChoice: string;
+  name: string;
+  size: 'large' | 'small';
+  textColor: string;
+}
+
+interface ColorSwatchStylesGroup {
+  [key: string]: FlattenSimpleInterpolation;
+}
+
+
+// STYLES
+const ColorSwatchSizeStyles: ColorSwatchStylesGroup = {
+  large: css`
+    .ColorSwatchPillContainer {
+      padding-top: 200px
+    }
+  `,
+  small: css`
+    .ColorSwatchPillContainer {
+      padding-top: 50px
+    }
+  `,
+}
+
+const StyledColorSwatch = styled.div`
+  background: ${props => props.backgroundColor };
+  color: ${props => props.textColor };
+  padding: 24px 16px 16px 16px;
+
+  .ColorSwatchName {
+    margin: 0;
+    font-weight: 600;
+  }
+
+  .ColorSwatchBackgroundColor {
+    margin: 0;
+  }
+
+  ${({ size = 'large' }: ColorSwatchProps) => ColorSwatchSizeStyles[size]}
+`;
+
+
+// JSX MARKUP
+export const ColorSwatch = ({
+  className,
+  backgroundColor,
+  contrastOnBlack,
+  contrastOnF1F1F1,
+  contrastOnFAFAFA,
+  contrastOnWhite,
+  copyChoice,
+  name,
+  size,
+  textColor
+}: ColorSwatchProps): React.ReactElement<any> => {
+
+  let copyToClipboardText = backgroundColor;
+
+  if (copyChoice == 'variableName') {
+    copyToClipboardText = name;
+  }
+
+  return (
+
+    <CopyToClipboard text={copyToClipboardText}>
+      <StyledColorSwatch
+        className={classNames(className)}
+        backgroundColor={backgroundColor}
+        textColor={textColor}
+        size={size}
+      >
+       <p class="ColorSwatchName">{name}</p>
```

Comment:
+1 on @brandonlilly's comment, but also: I built out a new component called Typography that can help do all the text formatting automatically. I'm going to dump this here with no explanation to see how effective Stories and IDE autocomplete is at telling a developer how to use it. If you hit turbulence using this, then blow me up immediately and that's what will inform what we document.

Reply:
@sharkparty the `Typography` component looks cool. Looking at the stories and the interface it seems intuitive to me how to use. I am running into 2 difficulties with it though.

1. I am setting the text color via a prop in the form of a hex code because it is dynamic based on color contrast. I have this prop available to pass in to `Typography` but it looks like its interface does not accept hex code values (also this seems like an edge case to me).

2. I'm wanting to zero out margin (like a normalize). Was that an intentional choice to leave that out?

I don't think either of these are worth making changes to `Typography` at the moment, but perhaps they are reasons for using classes for now, and I will use `Typography` for other simpler elements. What do you think?